checkoff

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+import argparse
+import json
+import subprocess
+import sys
+
+from ocflib.account.search import user_exists
+
+
+# Format of JSON file should be:
+# {
+#     "roster": [<list of OCF usernames of students>],
+#     "users": {
+#         "username1": [<list of labs checked off>],
+#         ...
+#     },
+#     "labs": {
+#         "a1": [<list of usernames checked off],
+#         ...
+#     },
+# }
+
+DATA_PATH = 'checkoff_data.json'
+
+
+def checkoff(args):
+    user = args.user
+    lab = args.lab
+    _check_lab_id(lab)
+
+    data = _load_data()
+    roster = data['roster']
+    users = data['users']
+    labs = data['labs']
+
+    if user not in roster:
+        print('{} not in class roster'.format(user))
+        return 1
+
+    if user in users:
+        if lab in users[user]:
+            print('{} already checked off for lab {}'.format(user, lab))
+            return 1
+
+        users[user].append(lab)
+    else:
+        users[user] = [lab]
+
+    if lab in labs:
+        labs[lab].append(user)
+    else:
+        labs[lab] = [user]
+
+    ans = input('OK to check off {} for lab {}? [y/N]\n'.format(user, lab))
+
+    if not ans or ans[0].lower() != 'y':
```

Comment:
Also I think this condition is a bit confusing because of the multiple negatives, maybe do `if not (ans and ans.lower() == 'y'):` or maybe make this a `if`/`else`? It's not too bad as it is, but if we added more to it, it would get a lot more confusing and it's bit harder to read right now.

Reply:
Yeah, I think `ans.lower() != 'y'` is sufficient.